google/cloud/aiplatform/metadata/execution.py

```diff
@@ -142,18 +142,110 @@ def create(
             Execution: Instantiated representation of the managed Metadata Execution.
 
         """
-        self = cls._empty_constructor(
-            project=project, location=location, credentials=credentials
+        # Add User Agent Header for metrics tracking if one is not specified
+        # If one is alreayd specified this call was initiated by a sub class.
+        if not base_constants.USER_AGENT_SDK_COMMAND:
+            base_constants.USER_AGENT_SDK_COMMAND = (
+                "aiplatform.metadata.execution.Execution.create"
```

Comment:
So that you don't have to hard code all of these module name strings, I wonder if you could save the prefix to a variable and store it in `constants/base.py` using:

```
user_agent_sdk_prefix = f"{self.__module__}.{self.__class__.__name__}."
```

In this case I think that would give you `"aiplatform.metadata.execution.Execution."`. And then you'd append the method name to that prefix. I'm not sure if there's a way to get the current method name, but if there is you could use that here too.

Reply:
good question, we experimented with using a flexible prefix as you mentioned however it will result in too many variations based on the calling subclass, for example for Schema classes depending on the artifact type the path will be different such as ``` sdk_command/google.cloud.aiplatform.metadata.schema.system.artifact_schema.Metrics.create``` if the create is called from Metrics schema subclass versus ```sdk_command/google.cloud.aiplatform.metadata.schema.system.artifact_schema.Model.create```

Based on offline discussion with Jialu I think for now we will keep the metadata implementation limited to the hard coded string. 

That being said for long term maintenance using a shared method to set the name would be much easier.